app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -402,6 +404,35 @@ public void setCurrentSession(int aId) {
             mCurrentSession.setActive(true);
     }
 
+    public String setRegion(String aRegion) {
+        Log.d(LOGTAG, "SessionStore setRegion: " + aRegion);
+        String result = "worldwide";
+        if (mCurrentSession == null || aRegion == null) {
+            return result;
+        }
+        State state = mSessions.get(mCurrentSession.hashCode());
+        if (state == null) {
+            return result;
+        }
+        result = state.mRegion = aRegion.toLowerCase();
+        return result;
+    }
+
+    public String getHomeUri() {
+        String result = SessionStore.HOME_WITHOUT_REGION_ORIGIN;
+        if (mCurrentSession == null) {
```

Comment:
Region as a SessionStore field will remove all this session sanity checking logic as getHomeUri will always return the same region bound home uri regardless of the session state.